packages/flutter_test/lib/src/controller.dart

```diff
@@ -779,41 +810,142 @@ abstract class WidgetController {
     });
   }
 
+  /// Calls [debugPrint] with the given message.
+  ///
+  /// This is overridden by the WidgetTester subclass to use the test binding's
+  /// [TestWidgetsFlutterBinding.debugPrintOverride], so that it appears on the
+  /// console even if the test is logging output from the application.
+  @protected
+  void printToConsole(String message) {
+    debugPrint(message);
+  }
+
   // GEOMETRY
 
   /// Returns the point at the center of the given widget.
-  Offset getCenter(Finder finder) {
-    return _getElementPoint(finder, (Size size) => size.center(Offset.zero));
+  ///
+  /// If `warnIfMissed` is true (the default is false), then the
+  /// returned coordinate is checked to see if it would actually hit
+  /// the specified element, and if not, a warning is printed to the
+  /// console.
+  Offset getCenter(Finder finder, { bool warnIfMissed = false, String callee = 'getCenter' }) {
+    return _getElementPoint(finder, (Size size) => size.center(Offset.zero), warnIfMissed: warnIfMissed, callee: callee);
   }
 
   /// Returns the point at the top left of the given widget.
-  Offset getTopLeft(Finder finder) {
-    return _getElementPoint(finder, (Size size) => Offset.zero);
+  Offset getTopLeft(Finder finder, { bool warnIfMissed = false, String callee = 'getTopLeft' }) {
```

Comment:
`warnIfMissed` should also be documented on this method and the other ones below - maybe with a macro?

Reply:
done